code/datums/components/defiber.dm

```diff
@@ -0,0 +1,230 @@
+		G = H.get_ghost()
+		if(istype(G))
+			user.visible_message(span_warning("[icon2html(source, viewers(user))] \The [source] buzzes: Defibrillation failed. Patient's soul has almost departed, please try again."))
+			return
+		//We couldn't find a suitable ghost, this means the person is not returning
+		user.visible_message(span_warning("[icon2html(source, viewers(user))] \The [source] buzzes: Patient has a DNR."))
+		return
+
+	if(!H.client) //Freak case, no client at all. This is a braindead mob (like a colonist)
+		user.visible_message(span_warning("[icon2html(source, viewers(user))] \The [source] buzzes: Defibrillation failed. No soul detected."))
+		return
+
+	//At this point, the defibrillator is ready to work
+	if(!issynth(H))
+		H.adjustBruteLoss(-defib_heal_amt)
+		H.adjustFireLoss(-defib_heal_amt)
+		H.adjustToxLoss(-defib_heal_amt)
+		H.adjustOxyLoss(-H.getOxyLoss())
+		H.updatehealth() //Needed for the check to register properly
+
+	if(H.health <= H.get_death_threshold())
+		user.visible_message(span_warning("[icon2html(source, viewers(user))] \The [source] buzzes: Defibrillation failed. Vital signs are too weak, repair damage and try again."))
+		return
+
+	user.visible_message(span_notice("[icon2html(source, viewers(user))] \The [source] beeps: Defibrillation successful."))
+	H.set_stat(UNCONSCIOUS)
+	H.emote("gasp")
+	H.chestburst = 0 //reset our chestburst state
+	H.regenerate_icons()
+	H.reload_fullscreens()
+	H.flash_act()
+	H.apply_effect(10, EYE_BLUR)
+	H.apply_effect(10, PARALYZE)
+	H.handle_regular_hud_updates()
+	H.updatehealth() //One more time, so it doesn't show the target as dead on HUDs
+	H.dead_ticks = 0 //We reset the DNR time
+	REMOVE_TRAIT(H, TRAIT_PSY_DRAINED, TRAIT_PSY_DRAINED)
+	GLOB.round_statistics.total_human_revives[H.faction]++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "total_human_revives[H.faction]")
+	to_chat(H, span_notice("You suddenly feel a spark and your consciousness returns, dragging you back to the mortal plane."))
+
+	if(CHECK_BITFIELD(H.status_flags, XENO_HOST))
+		var/obj/item/alien_embryo/friend = locate() in H
+		START_PROCESSING(SSobj, friend)
+
+	notify_ghosts("<b>[user]</b> has brought <b>[H.name]</b> back to life!", source = H, action = NOTIFY_ORBIT)
+
+
+//helper proc
+/mob/living/proc/get_ghost()
+	if(client) //Let's call up the correct ghost!
+		return null
+	for(var/mob/dead/observer/ghost AS in GLOB.observer_list)
+		if(!ghost) //Observers hard del often so lets just be safe
+			continue
+		if(isnull(ghost.can_reenter_corpse))
+			continue
+		if(ghost.can_reenter_corpse.resolve() != src)
+			continue
+		if(ghost.client)
+			return ghost
+	return null
+
+
+/mob/living/carbon/human/proc/has_working_organs()
+	var/datum/internal_organ/heart/heart = internal_organs_by_name["heart"]
+
+	if(!heart || heart.organ_status == ORGAN_BROKEN || !has_brain())
+		return FALSE
+
+	return TRUE
```

Comment:
Thy is this in this foile this looks like it should be in helpers

Also missing autodoc

Reply:
it was in the defib file where I copied it from, where should it be placed? also autodoc you mean documentation or?